app/services/doc_auth/mock/result_response.rb

```diff
@@ -1,16 +1,13 @@
 module DocAuth
   module Mock
-    class ResultResponseBuilder
+    class ResultResponse < DocAuth::Response
```

Comment:
In retrospect, maybe a builder would be more preferable here? At least in the sense that the constructor signature is very different between this class and the base `DocAuth::Response`.

That would obviously regress on the intended benefit of having `attention_with_barcode?` as a guaranteed class method, but maybe there's another way we could infer that from the base class.